src/zenml/logging/step_logging.py

```diff
@@ -78,34 +88,110 @@ def prepare_logs_uri(
         artifact_store.makedirs(logs_base_uri)
 
     # Delete the file if it already exists
-    logs_uri = os.path.join(logs_base_uri, f"{log_key}.log")
-    if artifact_store.exists(logs_uri):
+    logs_uri_folder = os.path.join(logs_base_uri, log_key)
+    if artifact_store.exists(logs_uri_folder):
         logger.warning(
-            f"Logs file {logs_uri} already exists! Removing old log file..."
+            f"Logs directory {logs_uri_folder} already exists! Removing old log directory..."
         )
-        artifact_store.remove(logs_uri)
-    return logs_uri
+        artifact_store.remove(logs_uri_folder)
+
+    artifact_store.makedirs(logs_uri_folder)
+    return logs_uri_folder
+
+
+def fetch_logs(
+    zen_store: "BaseZenStore",
+    artifact_store_id: Union[str, UUID],
+    logs_uri: str,
+    offset: int = 0,
+    length: Optional[int] = 1024 * 1024 * 16,  # Default to 16MiB of data
+) -> str:
+    """Fetches the logs from the artifact store.
+
+    Args:
+        zen_store: The store in which the artifact is stored.
+        artifact_store_id: The ID of the artifact store.
+        logs_uri: The URI of the artifact.
+        offset: The offset from which to start reading.
+        length: The amount of bytes that should be read.
+
+    Returns:
+        The logs as a string.
+
+    Raises:
+        DoesNotExistException: If the artifact does not exist in the artifact
+            store.
+    """
+    artifact_store = _load_artifact_store(artifact_store_id, zen_store)
+    if logs_uri.endswith(LOGS_EXTENSION):
+        return str(
+            _load_file_from_artifact_store(
+                logs_uri,
+                artifact_store=artifact_store,
+                mode="rb",
+                offset=offset,
+                length=length,
+            ).decode()
+        )
+    else:
+        files = artifact_store.listdir(logs_uri)
+        if len(files) == 1:
+            return str(
+                _load_file_from_artifact_store(
+                    os.path.join(logs_uri, str(files[0])),
+                    artifact_store=artifact_store,
+                    mode="rb",
+                    offset=offset,
+                    length=length,
+                ).decode()
+            )
+        else:
+            files.sort()
```

Comment:
The only thing that this currently does not handle is negative offsets, which I think are important as the plan is for the frontend to display the tail of the logging file first until the user scrolls up. Displaying the tail will probably be handled by calling `/logs?offset=-1000&length=1000` to fetch the latest 1000 bytes for example

Reply:
reworked